### PR TITLE
Add UserBanningEvent (Universal)

### PR DESCRIPTION
**Pull request by 01-Feli**
Please tell me if I did all the things correctly.

### CODE REVIEW

File: framework/OpenMod.Core/Users/Events/IUserBanningEvent.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using OpenMod.API.Eventing;
+using System;
+
+namespace OpenMod.Core.Users.Events
+{
+    public interface IUserBanningEvent : IUserEvent, ICancellableEvent
```
Comment (DiFFoZ) [inline framework/OpenMod.Core/Users/Events/IUserBanningEvent.cs:6]: It should be moved to OpenMod.Extensions.Games.Abstractions with IPlayerBanningEvent naming

File: unturned/OpenMod.Unturned/Users/UnturnedUserProvider.cs [hunk 2]
```diff
@@ -63,8 +63,46 @@ public class UnturnedUserProvider : IUserProvider, IDisposable
...
+                {
+                    ban = false;
+                }
+            });
+
+            reason = banReason;
+            duration = banTime;
+            shouldVanillaBan = ban;
+        }
+
```
Comment (DiFFoZ) [inline unturned/OpenMod.Unturned/Users/UnturnedUserProvider.cs:105]: Should be removed. Use instead [BansEventsListener](https://github.com/openmod/openmod/blob/main/unturned/OpenMod.Unturned/Players/Bans/Events/BansEventsListener.cs#L37)

File: unturned/OpenMod.Unturned/Users/Events/UnturnedUserBanningEvent.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            Reason = reason;
+            Duration = duration;
+        }
+
+        public string PunisherId { get; }
+        public string Reason { get; set; }
+        public DateTime Duration { get; set; }
+        public bool IsCancelled { get; set; }
+    }
+}
```
Comment (DiFFoZ) [inline unturned/OpenMod.Unturned/Users/Events/UnturnedUserBanningEvent.cs:26]: It should be removed. Use instead [UnturnedPlayerBanningEvent](https://github.com/openmod/openmod/blob/main/unturned/OpenMod.Unturned/Players/Bans/Events/UnturnedPlayerBanningEvent.cs)

File: framework/OpenMod.Core/Users/Events/IUserBanningEvent.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+using System;
+
+namespace OpenMod.Core.Users.Events
+{
+    public interface IUserBanningEvent : IUserEvent, ICancellableEvent
+    {
+        /// <summary>
+        /// Gets the punisher Id.
+        /// </summary>
+        public string PunisherId { get; }
```
Comment (DiFFoZ) [inline framework/OpenMod.Core/Users/Events/IUserBanningEvent.cs:11]: Add `PunisherType`

**Review by DiFFoZ — CHANGES_REQUESTED** [review]

File: framework/OpenMod.Core/Users/Events/IUserBanningEvent.cs (hunk 4 above)
Comment (01-Feli) [reply framework/OpenMod.Core/Users/Events/IUserBanningEvent.cs:11]: Why punisher type you mean IUser and the punisher id?

**Pull request closed without merging**